### PR TITLE
feat: add InvoiceNinja

### PR DESCRIPTION
**Pull request by olragon**
### ☑️ Self Check before Merge

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.

[create-env-file-from-sys-env.sh](https://gist.githubusercontent.com/olragon/e2f3ececd70b8c513ce6f768b47b6e5f/raw/0ce9188a0cbf1a9e6a7be4128419282b30492f49/create-env-file-from-sys-env.sh) script do 2 things
1. will read env vars from the system and overwritten `.env` file if needed
2. ensure directory `/var/www/app/docker-backup-public` exists.

Try hard for the InvoiceNinja app for One-Click-Apps experiment but I am not satisfied with current way, for example:
1. Where can I reliability put this (https://gist.githubusercontent.com/olragon/e2f3ececd70b8c513ce6f768b47b6e5f/raw/0ce9188a0cbf1a9e6a7be4128419282b30492f49/create-env-file-from-sys-env.sh) kind of script?
2. It's better to fix in https://github.com/invoiceninja/dockerfiles and https://github.com/invoiceninja/invoiceninja

### CODE REVIEW

File: public/v4/apps/invoiceninja.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            PHANTOMJS_PDF_GENERATION: $$cap_phantomjs_pdf_generation
+            PHANTOMJS_CLOUD_KEY: $$cap_phantomjs_cloud_key
+            BAK_PUBLIC_PATH: '/var/www/app/docker-backup-public'
+            NINJA_PROD: 'true'
+            ENV_LIST: 'NINJA_PROD,APP_NAME,APP_URL,APP_KEY,APP_DEBUG,MULTI_DB_ENABLED,DB_CONNECTION,DB_HOST1,DB_PORT1,DB_USERNAME1,DB_PASSWORD1,DB_DATABASE1,IS_DOCKER,PHANTOMJS_PDF_GENERATION,PHANTOMJS_CLOUD_KEY'
+        caproverExtra:
+            notExposeAsWebApp: 'true'
+            dockerfileLines:
+                - FROM invoiceninja/invoiceninja:$$cap_app_tag
+                - RUN curl -o create-env-file-from-sys-env.sh https://gist.githubusercontent.com/olragon/e2f3ececd70b8c513ce6f768b47b6e5f/raw/0ce9188a0cbf1a9e6a7be4128419282b30492f49/create-env-file-from-sys-env.sh
```
Comment (githubsaturn) [inline public/v4/apps/invoiceninja.yml:32]: It's okay for `create-env-file-from-sys-env.sh` script to be hosted on github gists as it's not mutable.

Alternatively, you can embed it here in the dockerfileLines, something like:
```
RUN echo "#!/bin/bash \n echo Hi" > script.sh && chmod +x script.sh
```

Your current method is better as it's more readable.

File: public/v4/apps/invoiceninja.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          defaultValue: user@example.com
+        - id: $$cap_user_password
+          label: User Password
+          defaultValue: $$cap_gen_random_hex(8)
+        - id: $$cap_app_key
+          label: App Key
+          defaultValue: $$cap_gen_random_hex(32)
+        - id: $$cap_app_debug
+          label: App Debug
+          defaultValue: 'true'
```
Comment (githubsaturn) [inline public/v4/apps/invoiceninja.yml:84]: Why is the default true for debug?

File: public/v4/apps/invoiceninja.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    variables:
+        - id: $$cap_app_tag
+          label: Invoice Ninja Tag
+          defaultValue: '5.1.53'
+          description: 'Check out their docker page for the valid tags https://hub.docker.com/r/invoiceninja/invoiceninja/tags'
+        - id: $$cap_user_email
+          label: User Email
+          defaultValue: user@example.com
+        - id: $$cap_user_password
+          label: User Password
```
Comment (githubsaturn) [inline public/v4/apps/invoiceninja.yml:77]:
```suggestion
          label: User Password (write this down as you won't be able to see it after deployment)
```

File: public/v4/apps/invoiceninja.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        - id: $$cap_db_name
+          label: MariaDB Database Name
+          defaultValue: 'ninja'
+        - id: $$cap_db_pass
+          label: MariaDB Database Password
+          defaultValue: $$cap_gen_random_hex(32)
+    instructions:
+        start: >-
+            Invoice Clients, Track Work-Time, Get Paid Online.
+            For more info visit https://www.invoiceninja.com/
```
Comment (githubsaturn) [inline public/v4/apps/invoiceninja.yml:115]: Would you be able to provide a link that explains the environmental variables such as `PHANTOMJS_PDF_GENERATION` and etc... ?

**Review by githubsaturn — COMMENTED** [review]
Thanks! Overall it looks good, just some minor things to get it working.

File: public/v4/apps/invoiceninja.yml (hunk 4 above)
Comment (olragon) [reply public/v4/apps/invoiceninja.yml:115]: Dig really hard to find this out https://github.com/invoiceninja/invoiceninja/issues/3390#issuecomment-797291670

Comment (olragon) [reply public/v4/apps/invoiceninja.yml:116]: I have just updated Caprover var's name for the self-explained reason.

**Comment by githubsaturn** [comment]
Thanks!

**Comment by smgthbusr** [comment]
Hello,
i just can get the One-Click App working(
Full default setup installs successfully but ..
- v5.1.53 somehow refuses to display the uploaded custom company logo in dashboard and invoices(pdf).
- after proper soft/software reboot of my Ubuntu server invoiceninja/invoiceninja-php just crashes and keeps creating new containers that also crash
So please guide me through some super secret working setup/installation
OR
Review/Update the code for the InvoiceNinja v5.1.70 version blease.
(I'm ready to provide some useful error details if needed)
Thanks

**Comment by githubsaturn** [comment]
Update: it's added here again: https://github.com/caprover/one-click-apps/pull/575

**Pull request merged**